ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -150,23 +172,24 @@ private void doCopyRetry(FileSystem sourceFs, List<ReplChangeManager.FileInfo> s
         }
       } else {
         // If destination file is missing, then retry copy
-        if (sourceFs.exists(srcPath)) {
+        if (srcFile.isUseSourcePath() && sourceFs.exists(srcPath)) {
           // If checksum does not match, likely the file is changed/removed, retry from CM path
           if (isSourceFileMismatch(sourceFs, srcFile)) {
             srcFile.setIsUseSourcePath(false);
           }
         } else {
-          if (srcFile.isUseSourcePath()) {
-            // Source file missing, then try with CM path
-            srcFile.setIsUseSourcePath(false);
-          } else {
-            // CM path itself is missing, cannot recover from this error
-            LOG.error("File Copy Failed. Both source and CM files are missing from source. "
-                    + "Missing Source File: " + srcFile.getSourcePath() + ", CM File: " + srcFile.getCmPath() + ". "
-                    + "Try setting higher value for hive.repl.cm.retain in source warehouse. "
-                    + "Also, bootstrap the system again to get back the consistent replicated state.");
-            throw new IOException("Both source and CM path are missing from source.");
-          }
+          // The file copy might have failed for network issue. So retry with cm path again if the
```

Comment:
We shall retain the old logic here as that doesn't have so many conditions and works fine.

Reply:
in old logic if the cm is disabled ..it will retry with cm path first time ..i think that is redundant